src/librustc_parse/parser/expr.rs

```diff
@@ -621,7 +621,44 @@ impl<'a> Parser<'a> {
                     }
                 }
             }
-        }
+        };
+
+        // Disallow postfix operators such as `.`, `?` or index (`[]`) after casts.
+        // Parses the postfix operator and emits an error.
+        let expr = type_result?;
```

Comment:
Style: Avoid the temporary here and put `return`'s where needed above (and consequently remove the `Ok(...)` rewrappings). Also, `expr.span` can be used in each place.

Reply:
Fixed the temporary/Ok wrapping!

As for `expr.span`, couldn't get that to work. We pass ownership of cast_expr into parse_dot_or_call_with_, and it seemed nicer to just save the span than to try to save the whole expr?